src/entities/Updates/model.ts

```diff
@@ -1,29 +1,56 @@
 import { Model } from 'decentraland-gatsby/dist/entities/Database/model'
 import { v1 as uuid } from 'uuid'
 
+import { VestingDates } from '../../clients/VestingData'
+import Time from '../../utils/date/Time'
+import { VestingStartDate } from '../Grant/types'
+
 import { UpdateAttributes, UpdateStatus } from './types'
 
 export default class UpdateModel extends Model<UpdateAttributes> {
   static tableName = 'proposal_updates'
   static withTimestamps = false
   static primaryKey = 'id'
 
-  static async createPendingUpdates(proposalId: string, duration: number) {
-    const updatesQuantity = duration
+  static async createPendingUpdates(
+    proposalId: string,
+    vestingDates: VestingDates,
+    preferredVestingStartDate: VestingStartDate
+  ) {
+    if (proposalId.length < 0) throw new Error('Unable to create updates for empty proposal id')
+
     const now = new Date()
+    const updatesQuantity = vestingDates.durationInMonths
+    const firstUpdateStartingDate = this.getFirstUpdateStartingDate(
+      vestingDates.vestingStartAt,
+      preferredVestingStartDate
+    )
+
+    await UpdateModel.delete<UpdateAttributes>({ proposal_id: proposalId, status: UpdateStatus.Pending })
```

Comment:
Isn't this deleting just one record?

Reply:
It deletes all records that match those conditions